ironic/drivers/modules/drac/raid.py

```diff
@@ -134,6 +135,56 @@ def list_physical_disks(node):
         raise exception.DracOperationError(error=exc)
 
 
+def is_raid_controller(node, raid_controller_fqdd):
+    """Find out if object's fqdd is for a raid controller or not
+
+    :param raid_controller_fqdd: The object's fqdd we are testing to see
+                                 if it is a raid controller or not.
+    :returns: boolean, True if the device is a RAID controller,
+              False if not.
+    """
+    client = drac_common.get_drac_client(node)
+
+    try:
+        return client.is_raid_controller(raid_controller_fqdd)
+    except drac_exceptions.BaseClientException as exc:
+        LOG.error('DRAC driver failed to get the list of raid controller '
+                  'for node %(node_uuid)s. Reason: %(error)s.',
+                  {'node_uuid': node.uuid, 'error': exc})
+
+        raise exception.DracOperationError(error=exc)
+
+
+def configure_physical_disk_conversion(node, mode,
+                                       controllers_to_physical_disk_ids=None):
+    """ Change physical to jbod to raid and vice-versa"""
+
+    list_unfinished_jobs = drac_job.list_unfinished_jobs(node)
+    if list_unfinished_jobs:
+        drac_job.wait_for_jobs_completion(node)
+
+    client = drac_common.get_drac_client(node)
+
+    try:
+        change_state_result = client.change_physical_disk_state(
+            mode, controllers_to_physical_disk_ids)
+        controllers = []
+        if change_state_result['commit_required_ids']:
```

Comment:
This code will need to be updated to use: https://review.opendev.org/#/c/654608/

Reply:
This function will not be needed. as you suggested new implementation to add substep after delete_configuration(). so i will remove this in next commit.